### PR TITLE
Script utility

### PR DESCRIPTION
**Pull request by rcooke-ast**
This PR contains a utility script that is designed to modularise all of the `scripts` (eventually). It takes an input `.pypeit` file to generate the ingredients about setup, frame types, and so forth.

I haven't generated any tests yet, because I wanted to get **_very broad feedback_** at first if:
(1) this is something that folks would find useful, and
(2) if I've gone about it in the right way.

The advantages (in my opinion) are as follows:
* Lots of scripts do similar things (e.g. load slits, or get the master dir/key), so having a single script that does these basic operations helps maintain the scripts a little better
* It's much easier to test, because the various utilities (such as loading the slit information) can be tested (just once).

### CODE REVIEW

File: pypeit/scripts/utils.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            mdir = self.par['calibrations']['master_dir']
+
+        if not os.path.exists(mdir):
+            mdir_base = os.path.join(os.getcwd(), os.path.basename(mdir))
+            msgs.warn('Master file dir: {0} does not exist. Using {1}'.format(mdir, mdir_base))
+            mdir = mdir_base
+        # Get the base name
+        file_base = self.get_basename(iFile=iFile)
+        ftdict = dict({file_base: 'science'})
+        fitstbl = PypeItMetaData(self.spectrograph, self.par,
```
Comment (profxj) [inline pypeit/scripts/utils.py:70]: Are you sure there are multiple scripts that need `fitstbl`?

I can only think of the `PypeIt` class and even its script doesn't use it.

File: pypeit/scripts/utils.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        # Load the spectrograph
+        spectrograph_name = cfg['rdx']['spectrograph']
+        self.spectrograph = load_spectrograph(spectrograph_name)
+        msgs.info('Loaded spectrograph {0}'.format(self.spectrograph.spectrograph))
+        spectrograph_cfg_lines = self.spectrograph.config_specific_par(fname).to_config()
+        # Load the parset
+        self.par = PypeItPar.from_cfg_lines(cfg_lines=spectrograph_cfg_lines, merge_with=self.cfg_lines)
+        return
+
+    def select_science_frame(self, use_first=False):
```
Comment (profxj) [inline pypeit/scripts/utils.py:101]: Is this just from the PypeIt file or `fitstbl`?  I think just the former as you would code differently for the `fitstbl`.

File: pypeit/scripts/utils.py (hunk 1 above)
Comment (rcooke-ast) [reply pypeit/scripts/utils.py:70]: So, I'm not actually storing the `fitstbl` here, I'm using it to generate the master key and master dir (based only on the input `.pypeit` file). I think this has to be loaded if I'm not passing a Master calibration frame as input?

File: pypeit/scripts/utils.py (hunk 2 above)
Comment (rcooke-ast) [reply pypeit/scripts/utils.py:101]: That's right, it's just reading the frame type from the `.pypeit` file.

File: pypeit/scripts/utils.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+from pypeit.metadata import PypeItMetaData
+from pypeit.core import procimg
+from pypeit import msgs
+from pypeit import slittrace
+from pypeit import masterframe
+
+
+class Utilities:
+    def __init__(self, args):
+        self.pypeit_file = args.file
```
[the remark this answers is not in the document]
Comment (rcooke-ast) [reply pypeit/scripts/utils.py:29]: I'm changing this to be transparent about the arguments. I'll add the docstrings now.

File: pypeit/scripts/utils.py (hunk 1 above)
Comment (profxj) [reply pypeit/scripts/utils.py:102]: Can't remember myself.  Not sure if @kbwestfall will know..

Comment (kbwestfall) [reply pypeit/scripts/utils.py:102]: I honestly don't remember either, but it does look like it's needed if you want to generate the master key correctly.

**Comment by rcooke-ast** [comment]
I think that's a fair point - We should avoid re-generating the `fitstbl`. Yet, it seems like storing it for just the `master_key` is a bit overkill. Given that all setups are pre-organised by `pypeit_setup`, can I just generate a `master_key`? i.e. grab the `Setup` letter from the `.pypeit` file, and then assume `master_key = "{0:s}_1_01".format(setup_letter)` for the rest?

**Comment by rcooke-ast** [comment]
Thanks - merged!

**Pull request merged**